src/realm/object-store/sync/app.cpp

```diff
@@ -1107,18 +1173,16 @@ void App::update_location_and_resend(Request&& request, UniqueFunction<void(cons
 
             // If the location info was updated, update the original request to point
             // to the new location URL.
-            auto comp = AppUtils::split_url(request.url);
-            if (!comp.is_ok()) {
-                throw Exception(comp.get_status());
-            }
-            request.url = self->get_host_url() + comp.get_value().request;
+            auto url = util::Uri::parse(request.url);
+            request.url =
+                util::format("%1%2%3%4", self->get_host_url(), url.get_path(), url.get_query(), url.get_frag());
 
             // Retry the original request with the updated url
-            self->m_config.transport->send_request_to_server(
-                std::move(request), [self = std::move(self), completion = std::move(completion)](
-                                        Request&& request, const Response& response) mutable {
-                    self->check_for_redirect_response(std::move(request), response, std::move(completion));
-                });
+            self->m_config.transport->send_request_to_server(request, [self = std::move(self),
+                                                                       completion = std::move(completion),
+                                                                       request](const Response& response) mutable {
```

Comment:
I had convinced myself that this was net the same number of copies but now I think that's wrong.

Reply:
I think so - when I initially did the changes for the redirect and being able to resend the request, I tried very hard to ensure we either passing by reference or using a move to avoid making extra copies of the request and response. Mathias provided some assistance with the `send_request_to_server()` implementation and using the `unique_ptr`.

Reply:
I've updated this to eliminate all of the copies of Request (and most of the moves, which also involve a memory allocation because of std::map).